ambry-account/src/main/java/com/github/ambry/account/MySqlAccountService.java

```diff
@@ -363,53 +281,61 @@ ExecutorService getScheduler() {
     // so subscribers can update their cache and avoid the database query.
-    super.publishChangeNotice();
+    accountMigrationService.publishChangeNotice();
+    if (config.enableNewDatabaseForMigration) {
+      accountMigrationServiceNew.publishChangeNotice();
+    }
   }
 
   @Override
   protected void onAccountChangeMessage(String topic, String message) {
-    if (!open.get()) {
-      // take no action instead of throwing an exception to silence noisy log messages when a message is received while
-      // closing the AccountService.
-      return;
-    }
-    try {
-      switch (message) {
-        case FULL_ACCOUNT_METADATA_CHANGE_MESSAGE:
-          logger.info("Processing message={} for topic={}", message, topic);
-          fetchAndUpdateCache();
-          logger.trace("Completed processing message={} for topic={}", message, topic);
-          break;
-        default:
-          accountServiceMetrics.unrecognizedMessageErrorCount.inc();
-          throw new RuntimeException("Could not understand message=" + message + " for topic=" + topic);
-      }
-    } catch (Exception e) {
-      logger.error("Exception occurred when processing message={} for topic={}.", message, topic, e);
-      accountServiceMetrics.remoteDataCorruptionErrorCount.inc();
-      accountServiceMetrics.fetchRemoteAccountErrorCount.inc();
+   accountMigrationService.onAccountChangeMessage(topic, message);
+   if (config.enableNewDatabaseForMigration) {
+     accountMigrationServiceNew.onAccountChangeMessage(topic, message);
+   }
+  }
+
+  @Override
+  public Set<Container> getContainersByStatus(Container.ContainerStatus containerStatus) {
+    return accountMigrationService.getContainersByStatus(containerStatus);
+  }
+
+  @Override
+  public void selectInactiveContainersAndMarkInStore(AggregatedAccountStorageStats aggregatedAccountStorageStats) {
+    accountMigrationService.selectInactiveContainersAndMarkInStore(aggregatedAccountStorageStats);
+    if (config.enableNewDatabaseForMigration) {
+      accountMigrationServiceNew.selectInactiveContainersAndMarkInStore(aggregatedAccountStorageStats);
     }
   }
 
   @Override
   public Collection<Container> updateContainers(String accountName, Collection<Container> containers)
       throws AccountServiceException {
-    try {
-      return super.updateContainers(accountName, containers);
-    } catch (AccountServiceException ase) {
-      if (needRefresh) {
-        // refresh and retry (with regenerated containerIds)
+      if (config.enableNewDatabaseForMigration) {
         try {
-          fetchAndUpdateCache();
+          if (mySqlAccountStoreNew == null) {
+            try {
+              mySqlAccountStoreNew = mySqlAccountStoreFactory.getMySqlAccountStoreNew();
+            } catch (SQLException e) {
+              logger.error("MySQL account store creation failed: {}", e.getMessage());
+              throw translateSQLException(e);
+            }
+          }
```

Comment:
why are we doing this? and can this logic got to migrationService.updateContainers?

Reply:
using supplier to handle lazy connection instead.